### PR TITLE
fix: save 2d array attribute with hdf5 matrix attr methods

### PR DESCRIPTION
**Pull request by bogovicj**
previously, 2d arrays would be serialized to json

### CODE REVIEW

File: src/main/java/org/janelia/saalfeldlab/n5/hdf5/N5HDF5Reader.java [hunk 1]
```diff
@@ -340,25 +340,49 @@ public <T> T getAttribute(String pathName, final String key, final Class<T> claz
...
+				return (T)reader.int64().getMatrixAttr(pathName, key);
+			else
+				return (T)reader.uint64().getMatrixAttr(pathName, key);
+		else if (type.isAssignableFrom(int[].class))
 			if (attributeInfo.isSigned())
 				return (T)reader.int32().getArrayAttr(pathName, key);
 			else
 				return (T)reader.uint32().getArrayAttr(pathName, key);
-		if (type.isAssignableFrom(short[].class))
+		else if (type.isAssignableFrom(int[][].class))
```
Comment (cmhulbert) [inline src/main/java/org/janelia/saalfeldlab/n5/hdf5/N5HDF5Reader.java:353]: Just to be sure, are `int[][]` assignable from `int[]`? If so, the 2D array variants should come before the 1D arrays.

Comment (bogovicj) [reply src/main/java/org/janelia/saalfeldlab/n5/hdf5/N5HDF5Reader.java:353]: :+1: good catch, yes we need to check that.

Comment (bogovicj) [reply src/main/java/org/janelia/saalfeldlab/n5/hdf5/N5HDF5Reader.java:353]: Turns out
```
int[].class.isAssignableFrom( int[][].class ); // returns false
int[][].class.isAssignableFrom( int[].class ); // returns false
```

**Comment by bogovicj** [comment]
merged here https://github.com/saalfeldlab/n5-hdf5/commit/18444bc5a3f07b4eb53ad25af41e7e91634b8ea3

**Pull request closed without merging**